### DISCUSSION TITLE
slash_substitute issue

### DISCUSSION THREAD

**Discussion by keskadaniel** [cantaloupe-project/cantaloupe#568] [General]
I deployed canteloupe on Azure and I had problem with slash encoding so I used slash_substitute property to workaround this. 
And it works but when i get info.json response from canteloupe, I get response with @id that contains path to file with encoded slases instead of my slash_substitute and my app  frontend can not use it. Could you please fix it to produce response with slash_substitute when sb use it?
Thank you in advance!

**Comment by adolski** [comment]
Hi @keskadaniel, I can't reproduce this in the latest version of Cantaloupe. What version are you using?

Here is my test:

#### cantaloupe.properties

```properties
slash_substitute = FOO
FilesystemSource.lookup_strategy = ScriptLookupStrategy
```

#### delegates.rb

```ruby
def filesystemsource_pathname(options = {})
  '/path/to/an/image'
end
```

Accessing: 

http://localhost:8182/iiif/2/somethingFOOsomething/info.json

Produces:

```json
  "@id": "http://localhost:8182/iiif/2/somethingFOOsomething"
```

**Reply by keskadaniel** [reply]
Oh sorry, I used v 3.2.2! Please tell me, from which version this slash_substitues works well? Cause now I have a problem with mounting files to v 5.0.5. Maybe earlier version will work. 
I use docker to run it.

**Reply by adolski** [reply]
I don't know what version this was fixed in, but I don't recommend using older versions.

Regarding this problem with mounting files, I don't know what is going on, but to go from 3.2.2 to 5.0.4, you'll have to work through the steps in the `UPGRADING.md` file, if you haven't yet. It might be easier to just start over with a new config file.

**Reply by keskadaniel** [reply]
Thanks, I changed properties file and version 5.0.5 works as expected! Thank you for suggestions, helped me a lot.
